src/Store/CommandStoreConnector.re

```diff
@@ -230,29 +230,24 @@ let start = (getState, contributedCommands) => {
         | None => EditorGroup.create()
         };
 
-      dispatch(EditorGroupAdd(newEditorGroup));
-
-      let split =
-        WindowTree.createSplit(
-          ~editorGroupId=newEditorGroup.editorGroupId,
-          (),
-        );
+      dispatch(AddSplit(direction, newEditorGroup.editorGroupId));
 
-      dispatch(AddSplit(direction, split));
+      // This needs to be dispatched after the split, since this will set the
+      // active editor group, which is then used as the target for the split.
+      dispatch(EditorGroupAdd(newEditorGroup));
```

Comment:
Another timing dependency caused by function-actions. This one was easily solved though, but the fix had to be applied in two places.

Reply:
Interesting. It'd be nice to remove this - I would imagine we could push one or both out of the effects and directly into the state update. Some untangling though needed.

Reply:
Yeah, something like that. I wanted to try to untangle it when replacing `WindowsStore` with `Feature_Layout.update`, but that's of course blocked by the other timing bug.